### ISSUE TITLE
Implement `InjectionStrategy.SETTER`

### ISSUE THREAD

**Issue by MLNW** [mapstruct/mapstruct#3229]
### Use case

It is possible that there is a dependency cycle between multiple mappers, e.g.:

```java
@Mapper(componentModel = "spring",  uses = { FooMapper.class })
public interface BarMapper {
    Bar toBo(BarDto dto);
    BarDto toDto(Bar obj) ;
}

@Mapper(componentModel = "spring",  uses = { BarMapper.class })
public interface ForMapper {
    Foo toBo(FooDto dto);
    FooDto toDto(Foo obj) ;
}
```

In that case newer versions of SpringBoot will fail at build time. This has been reported in https://github.com/mapstruct/mapstruct/issues/2741. https://github.com/mapstruct/mapstruct/issues/2741#issuecomment-1065410535 noted the preferred workaround, i.e., use `SETTER` injection instead of `FIELD` or `CONSTRUCTOR`.

Therefore, we should implement an additional `InjectionStrategy.SETTER` to work around issues like this.

### Generated Code

```java
@Mapper(componentModel = "spring",  uses = { FooMapper.class }, injectionStrategy = InjectionStrategy.SETTER)
public interface BarMapper {
    Bar toBo(BarDto dto);
    BarDto toDto(Bar obj) ;
}
```

```java
@Generated(...)
@Component
public class FooMapperImpl implements FooMapper {

    private BarMapper barMapper;

    ...

    public void setBarMapper(BarMapper barMapper) {
        this.barMapper = barMapper;
    }

}
```

### Possible workarounds

A work around is to use an external dependency described in the referenced issue: https://github.com/mapstruct/mapstruct/issues/2741#issuecomment-1067304830

### MapStruct Version

MapStruct 1.5.3.Final

**Comment by filiphr** [comment]
This makes sense as an enhancement @MLNW. The MapStruct team is busy with other things. However, if someone from the community is interested in working on this issue we are more than happy to help. @MLNW in case you are interested, you have the priority for working on this

**Comment by MLNW** [comment]
@filiphr I had a look through the code base and a short try of implementing this but in the end I wasn't able to find a way to add a non-mapping function to the generated mapper, as that is pretty much what needs to be done: add a `setFoo` function to the mapper.

Do you have an idea how to best accomplish this task?

**Comment by MLNW** [comment]
@filiphr I have added my progress as a draft PR. Maybe that will help in guiding me to the finish line.

**Comment by filiphr** [comment]
This was implemented with PR #3265 thanks @MLNW for your work on this and your first contribution to MapStruct
